packages/@aws-cdk/aws-stepfunctions/lib/fields.ts

```diff
@@ -1,8 +1,87 @@
 import { Token } from '@aws-cdk/core';
 import { findReferencedPaths, jsonPathString, JsonPathToken, renderObject } from './json-path';
 
+/**
+ * Extract a field from the State Machine data or context
+ * that gets passed around between states
+ *
+ * @see https://docs.aws.amazon.com/step-functions/latest/dg/amazon-states-language-paths.html
+ */
+export class JsonPath {
+  /**
+   * Instead of using a literal string, get the value from a JSON path
+   */
+  public static stringAt(path: string): string {
+    validateJsonPath(path);
+    return new JsonPathToken(path).toString();
+  }
+
+  /**
+   * Instead of using a literal string list, get the value from a JSON path
+   */
+  public static listAt(path: string): string[] {
+    // does not apply to task context
+    validateDataPath(path);
+    return Token.asList(new JsonPathToken(path));
+  }
+
+  /**
+   * Instead of using a literal number, get the value from a JSON path
+   */
+  public static numberAt(path: string): number {
+    validateJsonPath(path);
+    return Token.asNumber(new JsonPathToken(path));
+  }
+
+  /**
+   * Use the entire data structure
+   *
+   * Will be an object at invocation time, but is represented in the CDK
+   * application as a string.
+   */
+  public static get entirePayload(): string {
+    return new JsonPathToken('$').toString();
+  }
+
+  /**
+   * Determines if the indicated string is an encoded JSON path
+   *
+   * @param value string to be evaluated
+   */
+  public static isJsonPathString(value: string): boolean {
```

Comment:
This thing is for "internals". It doesn't do what the name suggests. Either move to a separate class or maybe rename to `isEncodedJsonPath` or something.

Reply:
good point, I got bitten once by this myself. I like `isEncodedJsonPath` as it reflects it's behaviour better